src/debugging-support-in-rustc.md

```diff
@@ -268,6 +268,45 @@ Focus is to let macros decide what to do. This can be achieved by having some ki
 that lets the macro tell the compiler where the line marker should be. This affects where you
 set the breakpoints and what happens when you step it.
 
+## Source file checksums in debug info
+
+Both DWARF and CodeView (PDB) support embedding a cryptographic hash of each source file that
+contributed the associated binary.
+
+The cryptographic hash can be used by a debugger to verify that the source file matches the
+executable. If the source file does not match, the debugger can provide a warning to the user.
+
+The hash can also be used to prove that a given source file has not been modified since it was
+used to compile an executable. Because MD5 and SHA1 both have demonstrated vulnerabilities,
+using SHA256 is recommended for this application.
+
+The Rust compiler stores the hash for each source file in the corresponding `SourceFile` in
+the `SourceMap`. The hashes of input files to external crates are stored in `rlib` metadata.
+
+A default hashing algorithm is set in the target specification. This allows the target to
+specify the best hash available, since not all targets support all hash algorithms.
+
+The hashing algorithm for a target can also be overridden with the `-Z source-file-checksum=`
```

Comment:
Questions for reviewers: Do we even want a command line option to override this? Or is rare enough that people wanting to override this could create a custom target specification?

Reply:
Would it be a good idea to spec this as a list of checksums, e.g. `-Z sha256,md5`, in case a target can support more than one?

Reply:
I would prefer to keep it as a single checksum for now. Both LLVM and CodeView (PDB) only support a single type of checksum. If a backend starts supporting multiple checksum algorithms, we can extend it then.